pkgs/applications/science/misc/root/default.nix

```diff
@@ -151,40 +151,18 @@ stdenv.mkDerivation rec {
   '';
 
   cmakeFlags = [
-    "-Drpath=ON"
     "-DCMAKE_INSTALL_BINDIR=bin"
     "-DCMAKE_INSTALL_LIBDIR=lib"
     "-DCMAKE_INSTALL_INCLUDEDIR=include"
     "-Dbuiltin_llvm=OFF"
-    "-Dbuiltin_freetype=OFF"
-    "-Dbuiltin_gtest=OFF"
-    "-Dbuiltin_nlohmannjson=OFF"
-    "-Dbuiltin_openui5=ON"
-    "-Dclad=ON"
-    "-Ddavix=ON"
-    "-Ddcache=OFF"
     "-Dfail-on-missing=ON"
-    "-Dfftw3=OFF"
     "-Dfitsio=OFF"
-    "-Dfortran=OFF"
     "-Dgnuinstall=ON"
-    "-Dimt=ON"
-    "-Dgviz=OFF"
-    "-Dhttp=ON"
     "-Dmysql=OFF"
-    "-Dodbc=OFF"
-    "-Dopengl=ON"
     "-Dpgsql=OFF"
-    "-Dpythia8=OFF"
-    "-Droot7=ON"
     "-Dsqlite=OFF"
-    "-Dssl=ON"
-    "-Dtmva=ON"
     "-Dtmva-pymva=OFF"
     "-Dvdt=OFF"
-    "-Dwebgui=ON"
-    "-Dxml=ON"
-    "-Dxrootd=ON"
```

Comment:
We don't want to rely on autoconfiguration for provided dependencies. The idea is that if the lookup is broken, we expect root to fail during configuration.

Reply:
Ah okay, maybe this is a misunderstanding! There is no autoconfiguration, thanks to the flag that is already added:
```
"-Dfail-on-missing=ON"
```

So if any feature that is already enabled by default like `xrootd` can't be built, the configuration will fail. There is zero difference between the default `xrootd=ON` or a manual `xrootd=ON`, but if the defaults would change because features are deprecated that would mean it has to be manually updated here.

More on this is explained here: https://root.cern/install/build_from_source/#external-and-built-in-dependencies

Reply:
Makes sense!